From da47c9b20031d3c7599d80d97436e4ab9d6ecc98 Mon Sep 17 00:00:00 2001
From: Dirk Hohndel <dirk@hohndel.org>
Date: Thu, 20 Jul 2023 02:07:46 +0000
Subject: [PATCH] allow running natively on x86_64

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>
---
 src/custompios | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/custompios b/src/custompios
index ed58934..1725d98 100755
--- a/src/custompios
+++ b/src/custompios
@@ -48,7 +48,10 @@ function execute_chroot_script() {
   cp "${CUSTOM_PI_OS_PATH}"/common.sh common.sh
   chmod 755 common.sh
   
-  if [ "$(uname -m)" != "armv7l" ] && [ "$(uname -m)" != "aarch64" ] && [ "$(uname -m)" != "arm64" ] ; then
+  if [ "$(uname -m)" == "$BASE_ARCH" ] ; then
+    echo "Building on the target architecture, not using qemu"
+    chroot . /bin/bash /chroot_script
+  elif [ "$(uname -m)" != "armv7l" ] && [ "$(uname -m)" != "aarch64" ] && [ "$(uname -m)" != "arm64" ] ; then
     if [ "$BASE_ARCH" == "armv7l" ]; then
       echo "Building on non-ARM device a armv7l system, using qemu-arm-static"
       if (grep -q gentoo /etc/os-release);then
-- 
2.34.1

